src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordObjEquals.cs

```diff
@@ -54,6 +54,13 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                 }
                 else
                 {
+                    if (_typedRecordEquals.ReturnType.SpecialType != SpecialType.System_Boolean)
+                    {
+                        // There is a signature mismatch, an error was reported elsewhere
+                        F.CloseMethod(F.ThrowNull());
+                        return;
```

Comment:
Would it be better if we avoided generating a method body in the error case?

Reply:
> Would it be better if we avoided generating a method body in the error case?

This is an error scenario. It is an established patter to ```CloseMethod``` like this when we are unable to produce something meaningful. I am following it. 

---
In reply to: [451900220](https://github.com/dotnet/roslyn/pull/45647#discussion_r451900220) [](ancestors = 451900220)